testing/integration/upgrade_fleet_test.go

```diff
@@ -0,0 +1,179 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+//go:build integration
+
+package integration
+
+import (
+	"context"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/elastic/elastic-agent-libs/kibana"
+
+	atesting "github.com/elastic/elastic-agent/pkg/testing"
+	"github.com/elastic/elastic-agent/pkg/testing/define"
+	"github.com/elastic/elastic-agent/pkg/testing/tools"
+	"github.com/elastic/elastic-agent/pkg/version"
+	"github.com/elastic/elastic-agent/testing/upgradetest"
+)
+
+// TestFleetManagedUpgrade tests that the build under test can retrieve an action from
+// Fleet and perform the upgrade. It does not need to test all the combinations of
+// versions as the standalone tests already perform those tests and would be redundant.
+func TestFleetManagedUpgrade(t *testing.T) {
+	info := define.Require(t, define.Requirements{
+		Stack: &define.Stack{},
+		Local: false, // requires Agent installation
+		Sudo:  true,  // requires Agent installation
+	})
+
+	ctx, cancel := context.WithCancel(context.TODO())
+	defer cancel()
+
+	// Start at the build version as we want to test the retry
+	// logic that is in the build.
+	startFixture, err := define.NewFixture(t, define.Version())
+	require.NoError(t, err)
+	err = startFixture.Prepare(ctx)
+	require.NoError(t, err)
+	startVersionInfo, err := startFixture.ExecVersion(ctx)
+	require.NoError(t, err)
+
+	// Upgrade to a different build but of the same version (always a snapshot).
+	// In the case there is not a different build then the test is skipped.
+	// Fleet doesn't allow a downgrade to occur, so we cannot go to a lower version.
+	sameVersion := define.Version()
+	if !strings.HasSuffix(sameVersion, "-SNAPSHOT") {
+		sameVersion += "-SNAPSHOT"
+	}
+	endFixture, err := atesting.NewFixture(
+		t,
+		sameVersion,
+		atesting.WithFetcher(atesting.ArtifactFetcher()),
+	)
+	require.NoError(t, err)
+	err = endFixture.Prepare(ctx)
+	require.NoError(t, err)
+	endVersionInfo, err := endFixture.ExecVersion(ctx)
+	require.NoError(t, err)
+	if startVersionInfo.Binary.Commit == endVersionInfo.Binary.Commit {
+		t.Skipf("Build under test is the same as the build from the artifacts repository (commit: %s)", startVersionInfo.Binary.Commit)
+	}
+
+	t.Logf("Testing Elastic Agent upgrade from %s to %s with Fleet...", define.Version(), endVersionInfo.Binary.String())
+
+	// start fixture gets the agent configured to use a faster watcher
+	err = upgradetest.ConfigureFastWatcher(ctx, startFixture)
+	require.NoError(t, err)
```

Comment:
Does this work for Fleet managed agents now?

Reply:
No it doesn't. I will remove this line.